### PR TITLE
Add char device to simple file system

### PR DESCRIPTION
**Pull request by gaotianyu1350**
Add some functions and attributes to INodeImpl and SimpleFileSystem in rcore-fs-sfs to support char device.

### CODE REVIEW

File: rcore-fs-sfs/src/lib.rs [hunk 1]
```diff
@@ -403,7 +451,13 @@ impl vfs::INode for INodeImpl {
         Ok(vfs::Metadata {
             dev: 0,
             inode: self.id,
-            size: disk_inode.size as usize,
+            size: match disk_inode.type_ {
+                FileType::File | FileType::SymLink => disk_inode.size as usize,
+                FileType::Dir => disk_inode.blocks as usize,
```
Comment (wangrunji0408) [inline rcore-fs-sfs/src/lib.rs]: Now the size of directory is `disk_inode.size` as well.

Comment (wangrunji0408) [reply rcore-fs-sfs/src/lib.rs]:
```suggestion
                FileType::Dir => disk_inode.size as usize,
```

**Comment by gaotianyu1350** [comment]
> LGTM.
> BTW, what's the usage of `link_inodeimpl`?

link_inodeimpl(device_file_name, inode_impl)

- `name: string`: name of the device file
- `inode_impl: &Arc<INodeImpl>`:  which represents the device file

eg: 

```rust
// add device to sfs
SFS.new_device_inode(1, STDIN.clone()) // STDIN is INode implementing stdin
// get a new INodeImpl of a device
let stdin_inodeimpl = SFS.new_inode_chardevice(1) // 1 is the device number
// link it to /dev
dev_inode_impl.link_inodeimpl("stdin", &stdin_inodeimpl) // dev_inode_impl is INodeImpl of "/dev"
```

**Comment by jiegec** [comment]
Okay, got it.

**Pull request merged**